### PR TITLE
Project detail

### PR DESCRIPTION
**Pull request by dereklieu**
Hooks up the majority of the project page, including timeline, KMI indicators, overview card.

Missing the Project Comparison and Related Projects features, but will fill those in separately.

### CODE REVIEW

File: app/assets/scripts/components/project-timeline.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    const start = Math.min(plannedStart, actualStart);
+    const end = Math.max(plannedEnd, actualEnd);
+
+    const scale = scaleTime().domain([new Date(start), new Date(end)]).range([0, 100]);
+
+    return (
+      <div className='timeline'>
+        <div className='timeline__unit'>
+          <h5>Current Progress</h5>
+          {actualStart && actualEnd && timeline(actualStart, actualEnd, scale)}
```
Comment (kamicut) [inline app/assets/scripts/components/project-timeline.js]: @dereklieu when this doesn't exist, should we show "No data"? Right now there's nothing displayed.

Comment (dereklieu) [inline app/assets/scripts/components/project-timeline.js]: Oh I actually think the whole `timeline__unit` div and all it's child elements should be hidden if we can't show it. I'll go and fix that.

Comment (dereklieu) [inline app/assets/scripts/components/project-timeline.js]: Hiding the actual date timeline when it doesn't exist. Can I assume that the proposed dates will always be there?

**Comment by kamicut** [comment]
@dereklieu unless you're going to do more work on this, good for merge.

**Pull request merged**